Tasks/WindowsMachineFileCopy/RoboCopyJob.ps1

```diff
@@ -171,13 +171,62 @@ param (
         Write-Verbose "Unable to get the path for net.exe. Net command will be 'net'"
         return 'net'
     }
+
+    function Create-DestinationDirectory(
+        [string]$path
+    )
+    {
+        $psCredentialObject = New-Object PSCredential($credential.UserName, (ConvertTo-SecureString $credential.Password -AsPlainText -Force))
+        $destPath = $path
+        $foundParentPath = $false
+        $isRoot = $false
+
+        Write-Verbose "Creating path to directory: $path"
+        while($destPath -and (-not $foundParentPath))
+        {
+            try
+            {
+                New-PSDrive -Name WFCPSDrive -PSProvider FileSystem -Root $destPath -Credential $psCredentialObject
```

Comment:
Ensure you test this in different OS versions. Check with @chshrikh on the matrix.

Reply:
1) win 2008 r2 2) win 2012 is key to test. Ensure you validate this with agent installed on os with PS3 version

Reply:
we should have const limit on how much sub folders we will create. 
1) add the logic to check the length of target path 
2) Add logic to validate number of sub folder we need to create and process/abort with proper error

Reply:
Why do we need to check the lengths of target paths or number of sub folders?